flask_peewee/tests/test_rest.py

```diff
@@ -527,6 +545,102 @@ def test_delete(self):
         f_obj = FModel.get(id=self.f1.id)
         self.assertEqual(f_obj.e, None)
 
+    def test_json_filtering(self):
+        # test that filtering on JSON fields works
+        resp = self.app.get('/api/jmodel?j_field.bar=car')
+        self.assertAPIJModels(
+            resp.get_json(),
+            JModel.select().where(JModel.j_field['bar'] == 'car').order_by(JModel.id))
+
+    def test_json_value(self):
+        self.create_test_models()
+        # test wrong pk
+        resp = self.app.get('/api/jmodel/9000/j_field/new')
+        self.assertEqual(resp.status_code, 404)
+
+        # test wrong field name
+        resp = self.app.get('/api/jmodel/9000/no_field/new')
+        self.assertEqual(resp.status_code, 404)
+
+        url = '/api/jmodel/%s' % self.j1.id
+
+        resp = self.app.get(url + '/j_field/foo')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == {'star': 'far'}
+
+        resp = self.app.get(url + '/j_field/foo/star')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == 'far'
+
+        resp = self.app.get(url + '/j_field/foo/tzar')
+        self.assertEqual(resp.status_code, 200)
+        assert resp.get_json() == None
```

Comment:
I'm unsure about these semantics. Would 404 be better and reserve 200 with value `None` for cases in which the property specified by the path is present but has a value of None?

Pushing on this a bit more, what would be expected for `GET .../j_field/foo/lets/dig/really/deeply/into/the/meaning/of/this/request`? What do we get today?